bliss.shy.js

```diff
@@ -353,7 +353,12 @@ extend($, {
 		env.xhr.open(env.method, env.url.href, env.async !== false, env.user, env.password);
 
 		for (var property in o) {
-			if (property in env.xhr) {
+			if (property === "upload") {
+				if (env.xhr.upload && typeof o[property] === "object") {
+					$.extend(env.xhr.upload, o[property]);
+				}
+			}
+			else if (property in env.xhr) {
```

Comment:
I would change `upload` inside the if to `property` (e.g. `env.xhr[property]`) so that if we need this behavior for other properties it's just a matter of changing the condition in the if. What do you think?

Reply:
I prefer to avoid abstractions unless they enable code reuse. I actually checked this, and `upload` is the only property in xhr which in turn is an object with settable properties. So it wouldn't add anything (unless a similar property is added to the xhr standard). If you still want this I could change it though.

Reply:
Nah, we can cross that bridge when we get to it :)
Merged, thanks!